src/classes/DynamicMapOfCountry.cls

```diff
@@ -0,0 +1,167 @@
+                	if(countryName.startsWith(startLetter) && stateName.startswith(startLetter)
+                       && districtName.startsWith(startLetter) ){
+                        listOfCombinations.add(countryName+','+stateName+ ','+districtName);
+                    }
+             	}  	  
+            }
+        }
+        if(!listOfCombinations.isEmpty()){
+            System.debug('Combination of Country, State and District:'+listOfCombinations);
+        }
+        aggregateQurey();
+        countryData(name);
+    }
+
+    /**   Description :   Method Update the Field of Country by using Aggregate Qurey  
+    *     Created By  :   Sagar Nirwal
+    *     Arguments   :   none
+    *     Return      :   void
+    *     Revision log:   V_1.0 - Created  - Sagar Nirwal - 14/06/2022
+    *                     V_1.1 - Modified - Sagar Nirwal - 14/06/2022 - Aggregate Qurey
+    **/
+    public static void aggregateQurey(){
+
+        //Aggregate Qurey------> Total District Of State   
+        Map<Id, State__c> mapOfTotalDistricts = new Map<Id, State__c>();                                       
+        for(AggregateResult aggre : [SELECT  State__c state, COUNT(Name) 
+                                                      FROM District__c 
+                                                      GROUP BY State__c]){
+            mapOfTotalDistricts.put(String.valueOf(aggre.get('state')),new State__c(Id = (Id)aggre.get('state'),
+                Total_District__c = (Decimal)aggre.get('expr0')));
+        }
+        if(!mapOfTotalDistricts.isEmpty()){
+            update mapOfTotalDistricts.values();
+        }
+        //Aggregate Qurey------> Total Sate AND District Of Country
+        Map<Id, Country__c> mapOfCountryField = new Map<Id, Country__c>();
+        for (AggregateResult agg : [SELECT Country__c coun, COUNT(Name), SUM(Total_District__c)
+                                                               FROM State__c 
+                                                               GROUP BY Country__c]){   
+            
+            totalState = (Decimal)agg.get('expr0'); 
+            totalDistrict = (Decimal)agg.get('expr1');
+            mapOfCountryField.put(String.valueOf(agg.get('coun')),new Country__c(Id = (Id)agg.get('coun'),
+                Total_States__c = (Decimal)agg.get('expr0'),
+                Total_District__c = (Decimal)agg.get('expr1')));
+        }
+        if(!mapOfCountryField.isEmpty()){
+            update mapOfCountryField.values();   
+        }
+    }
+    /**   Description :   JSON Data
+    *     Created By  :   Sagar Nirwal
+    *     Arguments   :   String
+    *     Return      :   void
+    *     Revision log:   V_1.0 - Created  - Sagar Nirwal - 14/06/2022
+    *                     V_1.1 - Modified - Sagar Nirwal - 14/06/2022 - Passing the argument 
+    **/    
+    public static void countryData(String conutryName){    
+        // JSON Data
+        
```

Comment:
Use SOQL on State and add filter of the country name and get all required fields of state from that query.